convert_test.go

```diff
@@ -174,3 +202,51 @@ func TestInTimezone(base *testing.T) {
 		})
 	}
 }
+
+// timezoneMetadata is a struct that contains timezone metadata for assertions
+// and translation across timezones. Intended to be used with `TestFromTime()`.
+type timezoneMetadata struct {
+	InTimezone *string
+	Name       *string
+	Offset     *int
+}
+
+// In translates a timestamp to an "in timezone" if one is set on this
+// metadata struct.
+func (tm timezoneMetadata) In(assert *testifyrequire.Assertions, t time.Time) time.Time {
+	if tm.InTimezone == nil {
+		return t
+	}
+
+	tz, err := time.LoadLocation(*tm.InTimezone)
+	assert.Nil(err)
+	return t.In(tz)
+}
+
+// ExpectedName returns the expected timezone name.
+func (tm timezoneMetadata) ExpectedName(t time.Time) string {
+	if tm.Name == nil {
+		return "UTC"
+	}
+
+	name := *tm.Name
+	tz := t.Location()
+	if name == "" && tz == time.Local {
+		name, _ = t.Zone()
+	}
```

Comment:
I realized when doing this that tests could fail on one machine and not the other. For example on my local machine which is in `TZ=America/Chicago`, if the parsed offset matches the local timezone, then the timezone is `time.Local` (not a `time.FixedZone("", offset)`)

Reply:
I made sure this worked in at least 3 timezones (3 named here):

```
$ TZ=America/Chicago go test -count=1 ./...
ok      github.com/hardfinhq/go-date    0.118s
$ TZ=America/New_York go test -count=1 ./...
ok      github.com/hardfinhq/go-date    0.109s
$ TZ=UTC go test -count=1 ./...
ok      github.com/hardfinhq/go-date    0.105s
```